python/cuml/cuml/accel/pytest_plugin.py

```diff
@@ -26,3 +33,89 @@ def pytest_load_initial_conftests(early_config, parser, args):
         raise RuntimeError(
             "An existing plugin has already loaded sklearn. Interposing failed."
         )
+
+
+def pytest_addoption(parser):
+    """Add command line option for xfail list file."""
+    parser.addoption(
+        "--xfail-list",
+        action="store",
+        help="Path to YAML file containing list of test IDs to mark as xfail",
+    )
+
+
+def create_version_condition(condition_str: str) -> bool:
+    """Evaluate a version condition immediately.
+
+    Args:
+        condition_str: String in format 'package[comparison]version'
+                      For example:
+                      - 'scikit-learn>=1.5.2'
+                      - 'numpy<2.0.0'
+                      - 'pandas==2.1.0'
+
+    Returns:
+        bool: True if the condition is met, False otherwise
+    """
+    if not condition_str:
+        return True
+
+    try:
+        req = Requirement(condition_str)
+        installed_version = version(req.name)
+        return req.specifier.contains(installed_version)
+    except Exception:
+        return False
+
+
+def pytest_collection_modifyitems(config, items):
+    """Apply xfail markers to tests listed in the xfail list file."""
+    # Import pytest lazily to avoid requiring it for normal cuml usage.
+    # pytest is only needed when running tests.
+    import pytest
+
+    xfail_list_path = config.getoption("xfail_list")
+    if not xfail_list_path:
+        return
+
+    xfail_list_path = Path(xfail_list_path)
+    if not xfail_list_path.exists():
+        raise ValueError(f"Xfail list file not found: {xfail_list_path}")
+
+    xfail_list = yaml.safe_load(xfail_list_path.read_text())
+
+    if not isinstance(xfail_list, list):
+        raise ValueError("Xfail list must be a list of test entries")
+
+    # Convert list of dicts into dict mapping test IDs to lists of xfail configs
+    xfail_configs = defaultdict(list)
+    for entry in xfail_list:
+        if not isinstance(entry, dict):
+            raise ValueError("Xfail list entry must be a dictionary")
+        if "id" not in entry:
+            raise ValueError("Xfail list entry must contain an 'id' field")
+
+        test_id = entry["id"]
+        condition = True
+        if "condition" in entry:
+            condition = create_version_condition(entry["condition"])
+
+        config = {
+            "reason": entry.get("reason", "Test listed in xfail list"),
```

Comment:
Scikit-learn might have their own xfail markers on some tests. I'd find it a bit clearer if the reason we used clearly mentioned `cuml` in the common case. Something like

```suggestion
            "reason": entry.get("reason", "Test xfailed under `cuml.accel`"),
```